### PR TITLE
Add GetXPath method to Element

### PR DESCRIPTION
**Pull request by alexferrari88**
This PR adds the `GetXPath` method to the `Element` struct.
It allows retrieving the XPath of an element by running the same JS used by Chrome Developer Tools. 

It can be useful if one needs to retrieve the XPath of elements that can only be selected by text or tag. It is based on [playwright-dom-path](https://github.com/alexferrari88/playwright-DOMPath) _(full disclosure: this is a package of mine)_.

On StackOverflow, you can find many people asking how to do this in playwright or selenium so it would be handy to have it in rod as well.

### CODE REVIEW

File: element.go [hunk 1]
```diff
@@ -194,10 +194,9 @@ func (el *Element) Interactable() (pt *proto.Point, err error) {
 // A 4-gon is not necessary a rectangle. 4-gons can be apart from each other.
 // For example, we use 2 4-gons to describe the shape below:
 //
-//       ____________          ____________
-//      /        ___/    =    /___________/    +     _________
-//     /________/                                   /________/
-//
+//	  ____________          ____________
+//	 /        ___/    =    /___________/    +     _________
+//	/________/                                   /________/
```
Comment (ysmood) [inline element.go:199]: I think this change is not related to this PR, can we remove it?

File: element.go [hunk 2]
```diff
@@ -701,3 +700,13 @@ func (el *Element) Equal(elm *Element) (bool, error) {
...
...me; } if (left.nodeType === right.nodeType) { return true; } const leftType = left.nodeType === Node.CDATA_SECTION_NODE ? Node.TEXT_NODE : left.nodeType; const rightType = right.nodeType === Node.CDATA_SECTION_NODE ? Node.TEXT_NODE : right.nodeType; return leftType === rightType; } const parentNode = node.parentNode; const siblings = parentNode ? parentNode.children : null; if (!siblings) { return 0; } let hasSameNamedElements; for (let i = 0; i < siblings.length; ++i) { if (areNodesSimilar(node, siblings[i]) && !(siblings[i] === node)) { hasSameNamedElements = true; break; } } if (!hasSameNamedElements) { return 0; } let ownIndex = 1; for (let i = 0; i < siblings.length; ++i) { if (areNodesSimilar(node, siblings[i])) { if (siblings[i] === node) { return ownIndex; } ++ownIndex; } } return -1; }; const node = this; if (node.nodeType === Node.DOCUMENT_NODE) { return '/'; } const steps = []; let contextNode = node; while (contextNode) { const step = xPathValue(contextNode, optimized); if (!step) { break; } steps.push(step); if (step.optimized) { break; } contextNode = contextNode.parentNode; } steps.reverse(); return (steps.length && steps[0].optimized ? '' : '/') + steps.join('/');}"
```
Comment (ysmood) [inline element.go:706]: Can you add this helper to this file:

https://github.com/go-rod/rod/blob/master/lib/js/helper.js

**Review by ysmood — CHANGES_REQUESTED** [review]
Looks good to me, it just needs some polishments.

**Comment by alexferrari88** [comment]
I have put the function in the helper

**Comment by alexferrari88** [comment]
I see that after you synced this with master, the test in now failing. I'm working on it to fix it.

**Review by ysmood — CHANGES_REQUESTED** [review]
The eslint failed in the CI, please fix it: https://github.com/go-rod/rod/runs/8046314554?check_suite_focus=true#step:5:40

**Comment by alexferrari88** [comment]
> The eslint failed in the CI, please fix it: https://github.com/go-rod/rod/runs/8046314554?check_suite_focus=true#step:5:40

done

**Review by ysmood — CHANGES_REQUESTED** [review]
The PR looks good to me, can you pull the latest maser and resolve the conflict?

**Comment by alexferrari88** [comment]
> The PR looks good to me, can you pull the latest maser and resolve the conflict?

done!

**Review by ysmood — CHANGES_REQUESTED** [review]
It's not in sync with the master, can you read the errors in the github check below?

**Comment by alexferrari88** [comment]
As for the other PR, I have now merged it, sorry

**Comment by alexferrari88** [comment]
I have rebased on master, solved the conflict, and force pushed it. It should be good now.

**Review by ysmood — CHANGES_REQUESTED** [review]
Can you fix the CI error first?

**Comment by alexferrari88** [comment]
any suggestion on how I can make the method GetXPath return an error for testing purposes? So that I can get 100% test coverage

**Comment by ysmood** [comment]
Use the stub to retuen eval error, read the commemt of stub, it has examples

**Comment by alexferrari88** [comment]
> Use the stub to retuen eval error, read the commemt of stub, it has examples

I wanted indeed to use MockClient.StubError but I cannot see what I can pass as proto.Request in this case. I looked for examples in which you used MustEval but that didn't help...

**Comment by alexferrari88** [comment]
> Use the stub to retuen eval error, read the commemt of stub, it has examples

I wanted indeed to use MockClient.StubError but I cannot see what I can pass as proto.Request in this case. I looked for examples in which you used MustEval but that didn't help...

**Comment by alexferrari88** [comment]
Finally everything seems to be ok!

**Review by ysmood — APPROVED** [review]
Nice job!

**Comment by ysmood** [comment]
But it still has some conflict with the master branch, can you resolve it?

**Review by ysmood — CHANGES_REQUESTED** [review]
But it still has some conflict with the master branch, can you resolve it?

**Review by ysmood — APPROVED** [review]

**Pull request merged**